_posts/2018-12-29-dask-on-hpc.md

```diff
@@ -0,0 +1,191 @@
+
+
+### 6.  Cost and Collaboration
+
+Dask is free and open source, which means we do not have to rebalance our budget and staff to address the new immediate need of data analysis tools.
+We don't have to pay for licenses, and we have the ability to make changes to the code when necessary.  The HPC community has good representation among Dask developers.  It's easy for us to participate and our concerns are well understood.
+
+<!-- WR: Mention quick response to new use cases / demands as another benefit of the collaborative
+approach?  And hint towards dask-mpi here? -->
+
+
+# What needs work
+
+### 1.  Heterogeneous resources handling
+
+Often we want to include different kinds of HPC nodes in the same deployment.
+This includes situations like the following:
+
+-  Workers with low or high memory,
+-  Workers with GPUs,
+-  Workers from different node pools.
+
+Dask provides some support for this heterogeneity already, but not enough.
+We see two major opportunities for improvement.
+
+-  Tools like Dask-Jobqueue should make it easier to manage multiple worker
+   pools within the same cluster.  Currently the deployment solution assumes
+   homogeneity.
+-  It should be easier for users to specify which parts of a computation
+   require different hardware.  The solution today works, but requires more
+   detail from the user than is ideal.
+
+
+### 2.  Coarse-Grained Diagnostics and History
+
+Dask provides a number of profiling tools that deliver real-time diagnostics at the individual task-level, but there is no way today to analyze or profile your Dask application at a coarse-grained level, and no built-in way to track performance over long periods of time.
+
+Having more tools to analyze bulk performance would be helpful when making design decisions and future architecture choices.
+
+Having the ability to persist or store history of computations (`compute()` calls)
+and tasks executed on a scheduler could be really helpful to track problems and potential performance improvements.
+
+<!-- JJH: One tangible idea here would be a benchmarking suite that helps users make decisions about how to use dask most effectively.
+ -->
+
+### 3.  Scheduler Performance on Large Graphs
+
+HPC users want to analyze Petabyte datasets on clusters of thousands of large nodes.
+
+While Dask can theoretically handle this scale, it does tend to slow down a bit,
+reducing the pleasure of interactive large-scale computing. Handling millions of tasks can lead to tens of seconds latency before a computation actually starts.  This is perfectly fine for our Dask batch jobs, but tends to make the interactive Jupyter users frustrated.
+
+Much of this slowdown is due to task-graph construction time and centralized scheduling, both of which can be accelerated through a variety of means.  We expect that, with some cleverness, we can increase the scale at which Dask continues to run smoothly by another order of magnitude.
```

Comment:
`through a variety of means` this is a vague statement. Wouldn't it be worth being a bit more specific?

Reply:
Not sure this is the goal of this post. I think @mrocklin has some ideas on how Scheduler optimization could be done, but I'm not sure it's important to share them here.